File: atcoder/agc043_a.py
# https://atcoder.jp/contests/agc043/tasks/agc043_a
import sys
# input = sys.stdin.buffer.readline
def input(): return sys.stdin.readline().rstrip()
sys.setrecursionlimit(10 ** 7)

H, W = map(int, input().split())
S = [input() for _ in range(H)]
# print(S)
INF = 1001001001

dp = [[INF]*W for _ in range(H)]
if S[0][0] == '#':
    dp[0][0] = 1
else:
    dp[0][0] = 0

for r in range(H):
    for c in range(W):
        if r:
            if S[r][c] == '#' and S[r-1][c] == '.':
                dp[r][c] = min(dp[r][c], dp[r-1][c]+1)
            else:
                dp[r][c] = min(dp[r][c], dp[r-1][c])
        if c:
            if S[r][c] == '#' and S[r][c-1] == '.':
                dp[r][c] = min(dp[r][c], dp[r][c-1]+1)
            else:
                dp[r][c] = min(dp[r][c], dp[r][c-1])

print(dp[H-1][W-1])



# H, W = map(int, input().split())
# S = [input() for _ in range(H)]
# # print(S)
# dr = [ 1, 0]
# dc = [ 0, 1]
# ans = 1001001001

# def dfs(r,c,cnt,flg):
#     """
#     r, c
#     cnt: 連続黒の回数
#     flg: 1: 現在白（黒に来たらcnt+1）, 0: 現在黒
#     """
#     global ans
#     if S[r][c] == '#':
#         if flg:
#             cnt += 1
#             black_flg = 0
#     else:
#         flg = 1

#     if r==H-1 and c ==W-1:
#         ans = min(ans, cnt)
#         return

#     for dir in range(2):
#         nr = r + dr[dir]
#         nc = c + dc[dir]
#         if nr >= H or nc >= W: continue
#         dfs(nr, nc, cnt, flg)
#     return

# dfs(0,0,0,1)
# print(ans)

File: atcoder/agc043_a.cpp
// https://atcoder.jp/contests/agc043/tasks/agc043_a
#include<iostream>
// #include<algorithm>
// #include<string>
// #include<numeric>
// #include<vector>
// #include<map>
// #include<tuple>
// #include<set>
// #include<queue>
// #include<deque>
// #include<regex>
// #include<bitset>
// #include<iomanip>
// #include<complex>
// #include<stack>
// #include<functional>

#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for (int i = 0; i < (n); ++i)
#define drep(i,n) for(int i = (n-1); i >= 0; i--)
#define all(v) (v).begin(),(v).end()
template <class T> bool chmax(T &a, const T &b) { if (a < b) { a = b; return 1; } return 0; }
template <class T> bool chmin(T &a, const T &b) { if (b < a) { a = b; return 1; } return 0; }
template <class T> T gcd(T a, T b) { return b ? gcd(b, a % b) : a; }
template <class T> T lcm(T a, T b) { return a/gcd(a,b)*b; }
typedef pair<int, int> P;
typedef long long ll;
const int INF = 1001001001;
const ll LINF = 1001002003004005006ll;
const ll MOD = 1e9+7;


int main() {
	int h, w;
	cin >> h >> w;
    vector<string> s(h);
    rep(i,h) cin >> s[i];
    vector<vector<int>> dp(h, vector<int>(w, INF));
    if(s[0][0]=='#') dp[0][0] = 1;
    else             dp[0][0] = 0;
    rep(i,h) rep(j,w) {
        if (i) {
            if (s[i][j]=='#' && s[i-1][j]=='.') dp[i][j] = min(dp[i][j], dp[i-1][j] + 1);
            else                                dp[i][j] = min(dp[i][j], dp[i-1][j]);
        }
        if (j) {
            if (s[i][j]=='#' && s[i][j-1]=='.') dp[i][j] = min(dp[i][j], dp[i][j-1] + 1);
            else                                dp[i][j] = min(dp[i][j], dp[i][j-1]);
        }
    }
	cout << dp[h-1][w-1] << endl;
	return 0;
}
